Realm/RLMQueryUtil.mm

```diff
@@ -972,12 +1040,105 @@ void convert_null(T&& value, Fn&& fn) {
             break;
         case RLMPropertyTypeUUID:
             convert_null(value, [&](auto&& value) {
-                add_bool_constraint(type, operatorType, column.resolve<W<UUID>>(), value_of_type<UUID>(value));
+                add_bool_constraint(type, operatorType, column.resolve<W<UUID>>(),
+                                    value_of_type<UUID>(value));
             });
             break;
         case RLMPropertyTypeAny:
-            throwException(@"Unsupported predicate value type",
-                           @"Object type '%@' not supported", RLMTypeToString(type));
+            convert_null(value, [&](auto&& value) {
+                add_mixed_constraint(operatorType,
+                                     predicateOptions,
+                                     column.resolve<W<Mixed>>(),
+                                     value);
+            });
+    }
+}
+
+#pragma mark Mixed Constraints
+
+template<typename C, typename T>
+void QueryBuilder::add_mixed_constraint(NSPredicateOperatorType operatorType,
+                                        NSComparisonPredicateOptions predicateOptions,
+                                        Columns<C>&& column,
+                                        T value)
+{
+    if (auto i = RLMDynamicCast<NSNumber>(value)) {
+        if ([value objCType][0] == 'c') {
+            add_bool_constraint(RLMPropertyTypeAny, operatorType, column, value_of_type<Bool>(value));
+        }
+        else if (numberIsInteger(i)) {
+            add_numeric_constraint(RLMPropertyTypeAny, operatorType, column, value_of_type<Int>(value));
+        }
+        else if ([value objCType][0] == *@encode(float)) {
+            add_numeric_constraint(RLMPropertyTypeAny, operatorType, column, value_of_type<Float>(value));
+        }
+        else if ([value objCType][0] == *@encode(double)) {
+            add_numeric_constraint(RLMPropertyTypeAny, operatorType, column, value_of_type<Double>(value));
+        }
+    }
```

Comment:
I don't think we want any of this. While writing literally `"mixedProp > true"` would obviously be silly, `"mixedProp > %@", someMixedValue` is perfectly reasonable and should work regardless of what the mixed value happens to be.

Reply:
The reason why this is where is because `@NO` can represent an int. So while we can get rid of all this, doing so would not let us throw an exception if a user tried to do  `("mixedProp > %@), @NO)`

Reply:
Yes, I think that should be allowed. `>` is a supported operation for Mixed properties.

Reply:
Also without these conditionals for specific type information queries like `@"anyCol BETWEEN %@", @[@0, @5]` don't work because numbers like `@0` and `@1` appear as `BOOL`'s

Reply:
That shouldn't matter? The query engine appears to do the same thing for `col < Mixed(true)` and `col < Mixed(1)`.